### PR TITLE
Further refinement to transparency options

### PR DESCRIPTION
**Pull request by tupaschoal**
06302f7 - This undoes 9e1b24d, which apparently @Broxzier pushed by accident 2y ago and no one noticed 🤣 (likely because loading transparency options were broken on startup until #22659 

be1153b - Untoggling the invisible transparency would not untoggle the see-through, even though it might have been the one to set it. So now whenever unchecking the invisible, it also unchecks the see through. Similarly, if you had both see through and invisible checked and unchecked the see-through, it would remain invisible, so when disabling see through, it now disables invisible too.

4d2efa0 - Improves quick fix of #22670 to not affect sound on title scren

I think this makes it for an easier time using the window, but I'm open for suggestions.

### CODE REVIEW

File: src/openrct2-ui/windows/Main.cpp [hunk 1]
```diff
@@ -53,38 +53,41 @@ static Widget _mainWidgets[] = {
         }
 
     private:
-        void SetViewportFlags()
+        void SetViewportFlags(bool isTitleWindow)
         {
             viewport->flags |= VIEWPORT_FLAG_SOUND_ON;
-            if (Config::Get().general.InvisibleRides)
+            if (!isTitleWindow)
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/Main.cpp:59]: I'd prefer an early return over indenting here, tbh.

Comment (tupaschoal) [reply src/openrct2-ui/windows/Main.cpp]: Done

File: src/openrct2-ui/windows/Main.cpp [hunk 2]
```diff
@@ -53,9 +53,14 @@ static Widget _mainWidgets[] = {
         }
 
     private:
-        void SetViewportFlags()
+        void SetViewportFlags(bool isTitleWindow)
         {
             viewport->flags |= VIEWPORT_FLAG_SOUND_ON;
+            if (isTitleWindow)
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/Main.cpp:59]: Any particular reason over adding a parameter, as opposed to just testing for `gScreenFlags & SCREEN_FLAGS_TITLE_DEMO` here?

Comment (tupaschoal) [reply src/openrct2-ui/windows/Main.cpp:59]: No, just though it would make intent clearer, but I can change it.

**Review by AaronVanGeffen — APPROVED** [review]
I'm fine with these changes, but I have not been able to test them thoroughly. Would be good to have these addresses issues addressed in the upcoming release, though.

**Comment by cheweytoo** [comment]
This has again reset my in-game transparency settings to "everything invisible" on every saved park load, and the only way to fix this is to manually edit the config.ini: My changes made in game are not written to the config file.

**Comment by tupaschoal** [comment]
> This has again reset my in-game transparency settings to "everything invisible" on every saved park load, and the only way to fix this is to manually edit the config.ini: My changes made in game are not written to the config file.

Wait, if you open the transparency options at the bottom of the menu are they not toggled on? I tested this and it I cannot reproduce what you said. If you toggle the invisibility settings, they are saved and persist to other saves and sessions

**Comment by cheweytoo** [comment]
I use hotkeys to toggle the transparencies, and the settings do not persist:


https://github.com/user-attachments/assets/fffecbda-83f0-43c6-a3b1-557c13475b5b


This is recorded from develop 412e90d

**Comment by tupaschoal** [comment]
Ah, I see what the problem is. Likely the shortcuts do not trigger the same flow to change the settings on the config file. Can you file a new issue, please?

**Comment by sjoerddebruin** [comment]
We keep receiving reports on Discord of users that have invisible supports the moment they start the game for the first time since updating. This doesn't sound like intended behaviour to me.

**Comment by tupaschoal** [comment]
Discussion around this will be kept on #22682 22681

**Pull request merged**